library/src/main/java/com/novoda/downloadmanager/UnlinkedDataRemover.java

```diff
@@ -0,0 +1,21 @@
+package com.novoda.downloadmanager;
+
+class UnlinkedDataRemover {
+    
+    private final LocalFilesDirectory localFilesDirectory;
+    private final V2DatabaseFiles downloadFiles;
+
+    UnlinkedDataRemover(LocalFilesDirectory localFilesDirectory, V2DatabaseFiles downloadFiles) {
+        this.localFilesDirectory = localFilesDirectory;
+        this.downloadFiles = downloadFiles;
+    }
+
+    void remove() {
+        for (String fileName : localFilesDirectory.contents()) {
```

Comment:
Could do `RoomV2DatabaseFiles.fileNames` here instead of passing through the `V2DatabaseFiles` 🤔